src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_index_pattern/components/header/header.tsx

```diff
@@ -106,22 +111,43 @@ export const Header: React.FC<HeaderProps> = ({
               isInvalid={isInputInvalid}
               onChange={onQueryChanged}
```

Comment:
Should we be debouncing this?

While testing locally I noticed that we are firing a new HTTP request on every keystroke, which can be a lot if you are typing a longer index pattern name.

To be clear, this behavior existed before this PR, but since we are updating related behavior here I wonder if it'd be worth throwing this in a lodash `debounce`

Reply:
I attempted this but the debounce functionality was becoming troublesome to implement since it was conflicting with the code that appends a `*`. I'll try to address in a separate PR. If nothing else, the calls to `_resolve/indices` are much more performant than running queries as the code previously did.